tests/pos/TastyADT.scala

```diff
@@ -202,6 +197,8 @@ object tasty {
     object ErasedImplicitMethodType extends SpecializedMethodTypeCompanion
 
     case class TypeBounds(loBound: Type, hiBound: Type)
+    case class NoPrefix()
+    object NoPrefix extends NoPrefix
```

Comment:
Why both a class and an object for NoPrefix?

Reply:
To allow `NoPrefix` to be used in `|` types and be a singleton at the same time.